specification/evaluation-context/evaluation-context.json

```diff
@@ -0,0 +1,23 @@
+[
+    {
+        "id": "Requirement 3.1",
+        "clean id": "requirement_3_1",
+        "content": "The `evaluation context` structure MUST define a required `targeting key` field of type string, identifying the subject of the flag evaluation.",
+        "RFC 2119 keyword": "MUST",
+        "children": []
+    },
+    {
+        "id": "Requirement 3.2",
+        "clean id": "requirement_3_2",
+        "content": "The evaluation context MUST define the following optional fields: `email` (string), `first name` (string), `last name`(string), `name`(string), `ip`(string), `tz`(string), `locale`(string), `country code` (string), `timestamp`(date), `environment`(string), `application`(string), and `version`(string).",
+        "RFC 2119 keyword": "MUST",
+        "children": []
+    },
+    {
+        "id": "Requirement 3.3",
+        "clean id": "requirement_3_3",
+        "content": "The evaluation context MUST support the inclusion of custom fields, having keys of type `string`, and values of type `boolean | string | number | datetime`.",
```

Comment:
@toddbaert Can we add a `list` and `object` to these fields. 

For `datetime` is it expected to be RFC3339 or what specification are we looking to require here?

Reply:
For the `datetime` it seems like it would be clearer across language if we decided on which standard we expect, that way being explicit there's less room for subtle bugs across languages.

Also thoughts on adding `array` and `object` to the custom fields type? We will need support of those

Reply:
@InTheCloudDan In the java SDK, I'm planning to use [java.time.ZonedDateTime](https://docs.oracle.com/javase/8/docs/api/java/time/ZonedDateTime.html). Internally, we're representing it as a string, but that's opaque to the consumers of it.

Reply:
> For the datetime it seems like it would be clearer across language if we decided on which standard we expect, that way being explicit there's less room for subtle bugs across languages.

I've removed this field for now based on your other feedback @InTheCloudDan . If we decide we want to make it standard we can revisit this question.

> Also thoughts on adding array and object to the custom fields type? We will need support of those.

In the spirit of keeping this simple, are these really necessary? If they are, I'd suggest the language we use elsewhere: `structured data` since "object" and "array" don't always mean exactly the same things across languages (for example, when you say "object" you probably mean a "Map" in Java.